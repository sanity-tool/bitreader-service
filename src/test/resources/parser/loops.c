static int I;

extern void foo();

static void testWhile() {
    I = 0;
    while(I < 5) {
        foo();
        I++;
    }
}

static void testDoWhile() {
    I = 0;
    do {
        foo();
        I++;
    } while (I++);
}

static void testFor() {
    for (I = 0; I < 5; I++) {
        foo();
    }
}