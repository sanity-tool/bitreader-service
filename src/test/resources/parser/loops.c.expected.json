{
  "functions" : [ {
    "ref" : {
      "kind" : "Global",
      "index" : 0
    },
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMStore",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Global",
          "index" : 1
        }, {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 1
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMICmp",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 0
        }, {
          "kind" : "Global",
          "index" : 3
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 1
        }, {
          "kind" : "Block",
          "index" : -1
        }, {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 1
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Global",
          "index" : 4
        } ],
        "sourceRef" : 2
      }, {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 3
      }, {
        "kind" : "LLVMAdd",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 1
        }, {
          "kind" : "Global",
          "index" : 5
        } ],
        "sourceRef" : 3
      }, {
        "kind" : "LLVMStore",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 2
        }, {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 3
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : 1
        } ],
        "sourceRef" : 1
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMRet",
        "typeId" : 2,
        "sourceRef" : 4
      } ]
    } ],
    "entryBlockIndex" : 0
  }, {
    "ref" : {
      "kind" : "Global",
      "index" : 6
    },
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMStore",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Global",
          "index" : 1
        }, {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 5
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 6
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Global",
          "index" : 4
        } ],
        "sourceRef" : 7
      }, {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 8
      }, {
        "kind" : "LLVMAdd",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 1
        }, {
          "kind" : "Global",
          "index" : 5
        } ],
        "sourceRef" : 8
      }, {
        "kind" : "LLVMStore",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 2
        }, {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 8
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 9
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 9
      }, {
        "kind" : "LLVMAdd",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 0
        }, {
          "kind" : "Global",
          "index" : 5
        } ],
        "sourceRef" : 9
      }, {
        "kind" : "LLVMStore",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 1
        }, {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 9
      }, {
        "kind" : "LLVMICmp",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 0
        }, {
          "kind" : "Global",
          "index" : 1
        } ],
        "sourceRef" : 9
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 3
        }, {
          "kind" : "Block",
          "index" : -1
        }, {
          "kind" : "Block",
          "index" : 1
        } ],
        "sourceRef" : 9
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMRet",
        "typeId" : 2,
        "sourceRef" : 10
      } ]
    } ],
    "entryBlockIndex" : 0
  }, {
    "ref" : {
      "kind" : "Global",
      "index" : 7
    },
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMStore",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Global",
          "index" : 1
        }, {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 11
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 11
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 11
      }, {
        "kind" : "LLVMICmp",
        "typeId" : 5,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 0
        }, {
          "kind" : "Global",
          "index" : 3
        } ],
        "sourceRef" : 11
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 1
        }, {
          "kind" : "Block",
          "index" : -1
        }, {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 11
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMCall",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Global",
          "index" : 4
        } ],
        "sourceRef" : 12
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : -1
        } ],
        "sourceRef" : 13
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 11
      }, {
        "kind" : "LLVMAdd",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 0
        }, {
          "kind" : "Global",
          "index" : 5
        } ],
        "sourceRef" : 11
      }, {
        "kind" : "LLVMStore",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Instruction",
          "index" : 1
        }, {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 11
      }, {
        "kind" : "LLVMBr",
        "typeId" : 2,
        "operands" : [ {
          "kind" : "Block",
          "index" : 1
        } ],
        "sourceRef" : 11
      } ]
    }, {
      "instructions" : [ {
        "kind" : "LLVMRet",
        "typeId" : 2,
        "sourceRef" : 14
      } ]
    } ],
    "entryBlockIndex" : 0
  } ],
  "types" : [ {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 1 ]
  }, {
    "kind" : "LLVMFunctionTypeKind",
    "types" : [ 2 ]
  }, {
    "kind" : "LLVMVoidTypeKind"
  }, {
    "kind" : "LLVMIntegerTypeKind",
    "len" : 32
  }, {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 3 ]
  }, {
    "kind" : "LLVMIntegerTypeKind",
    "len" : 1
  }, {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 7 ]
  }, {
    "kind" : "LLVMFunctionTypeKind",
    "types" : [ 2 ]
  } ],
  "globals" : [ {
    "kind" : "LLVMFunctionValueKind",
    "name" : "testWhile",
    "typeId" : 0
  }, {
    "kind" : "LLVMConstantIntValueKind",
    "typeId" : 3,
    "intValue" : 0
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "I",
    "typeId" : 4,
    "operands" : [ {
      "kind" : "Global",
      "index" : 1
    } ]
  }, {
    "kind" : "LLVMConstantIntValueKind",
    "typeId" : 3,
    "intValue" : 5
  }, {
    "kind" : "LLVMFunctionValueKind",
    "name" : "foo",
    "typeId" : 6
  }, {
    "kind" : "LLVMConstantIntValueKind",
    "typeId" : 3,
    "intValue" : 1
  }, {
    "kind" : "LLVMFunctionValueKind",
    "name" : "testDoWhile",
    "typeId" : 0
  }, {
    "kind" : "LLVMFunctionValueKind",
    "name" : "testFor",
    "typeId" : 0
  } ],
  "sourceRefs" : [ {
    "file" : "parser/loops.c",
    "line" : 6
  }, {
    "file" : "parser/loops.c",
    "line" : 7
  }, {
    "file" : "parser/loops.c",
    "line" : 8
  }, {
    "file" : "parser/loops.c",
    "line" : 9
  }, {
    "file" : "parser/loops.c",
    "line" : 11
  }, {
    "file" : "parser/loops.c",
    "line" : 14
  }, {
    "file" : "parser/loops.c",
    "line" : 15
  }, {
    "file" : "parser/loops.c",
    "line" : 16
  }, {
    "file" : "parser/loops.c",
    "line" : 17
  }, {
    "file" : "parser/loops.c",
    "line" : 18
  }, {
    "file" : "parser/loops.c",
    "line" : 19
  }, {
    "file" : "parser/loops.c",
    "line" : 22
  }, {
    "file" : "parser/loops.c",
    "line" : 23
  }, {
    "file" : "parser/loops.c",
    "line" : 24
  }, {
    "file" : "parser/loops.c",
    "line" : 25
  } ]
}